src/lib/util/iheap.ml

```diff
@@ -97,16 +98,16 @@ let increase cmp s n =
 
 let filter s filt cmp =
   let j = ref 0 in
+  Vec.iter (fun elt ->
+      if filt elt then begin
+        Vec.set s.heap !j elt;
+        Vec.set s.indices elt !j;
+        incr j
+      end
+      else Vec.set s.indices elt (-1)
+    ) s.heap;
```

Comment:
It is often implicitly understood that combinators such as `Vec.iter` do not modify the underlying container, so please don't use `Vec.iter` when the iterated function modifies the container and use a `for` loop in that case.